src/lib.rs

```diff
@@ -135,7 +136,9 @@ impl IOSurface {
                                                   mem::transmute(self.as_concrete_TypeRef()),
                                                   0);
 
-            assert_eq!(gl_error, kCGLNoError);
+            if gl_error != kCGLNoErrer {
+                panic!(&*CStr::from_ptr(CGLErrorString(gl_error)).to_string_lossy());
```

Comment:
Do you mean returning a Result, and packing all the information into it so that rust-layers can read it?

Because the assertion creates a fatal error just like this panic does.